service/history/history_engine.go

```diff
@@ -379,17 +379,20 @@ func (e *historyEngineImpl) StartWorkflowExecution(
 	ctx context.Context,
 	startRequest *historyservice.StartWorkflowExecutionRequest,
 ) (resp *historyservice.StartWorkflowExecutionResponse, retError error) {
-	starter, err := startworkflow.NewStarter(
+	namespaceEntry, err := api.GetActiveNamespace(e.shardContext, namespace.ID(startRequest.GetNamespaceId()))
```

Comment:
Move it even higher? To `api`. The only reason `HistoryEngine` still exist is a lot of dependencies in tests.

Reply:
Hm, that would change the interface of the Engine which cascades throughout the tests. It might be better to revert this change instead.